extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageUtil.java

```diff
@@ -207,6 +245,136 @@ private static Optional<String> extractFlatBufferVersion(String method) {
+        BarrageSubscriptionRequest.addReverseViewport(metadata, reverseViewport);
+        metadata.finish(BarrageSubscriptionRequest.endBarrageSubscriptionRequest(metadata));
+
+        final FlatBufferBuilder wrapper = new FlatBufferBuilder();
+        final int innerOffset = wrapper.createByteVector(metadata.dataBuffer());
+        wrapper.finish(BarrageMessageWrapper.createBarrageMessageWrapper(
+                wrapper,
+                BarrageUtil.FLATBUFFER_MAGIC,
+                BarrageMessageType.BarrageSubscriptionRequest,
+                innerOffset));
+        return wrapper.sizedByteArray();
+    }
+
+    /**
+     * Create a snapshot request payload to be sent via DoExchange.
+     *
+     * @param ticketId the ticket id of the table to subscribe to
+     * @param options the barrage options
+     * @return the subscription request payload
+     */
+    public static byte[] createSnapshotRequestMetadataBytes(
+            @NotNull final byte[] ticketId,
+            @Nullable final BarrageSnapshotOptions options) {
+        return createSnapshotRequestMetadataBytes(ticketId, options, null, null, false);
+    }
+
+    /**
+     * Create a subscription request payload to be sent via DoExchange.
+     *
+     * @param ticketId the ticket id of the table to subscribe to
+     * @param options the barrage options
+     * @param viewport the viewport to subscribe to
+     * @param columns the columns to subscribe to
+     * @param reverseViewport whether to reverse the viewport
+     * @return the subscription request payload
+     */
+    static public byte[] createSnapshotRequestMetadataBytes(
+            @NotNull final byte[] ticketId,
+            @Nullable final BarrageSnapshotOptions options,
+            @Nullable final RowSet viewport,
+            @Nullable final BitSet columns,
+            final boolean reverseViewport) {
+
+        final FlatBufferBuilder metadata = new FlatBufferBuilder();
+
+        int colOffset = 0;
+        if (columns != null) {
+            colOffset = BarrageSnapshotRequest.createColumnsVector(metadata, columns.toByteArray());
+        }
+        int vpOffset = 0;
+        if (viewport != null) {
+            vpOffset = BarrageSnapshotRequest.createViewportVector(
+                    metadata, BarrageProtoUtil.toByteBuffer(viewport));
+        }
+        int optOffset = 0;
+        if (options != null) {
+            optOffset = options.appendTo(metadata);
+        }
+
+        final int ticOffset = BarrageSnapshotRequest.createTicketVector(metadata, ticketId);
+        BarrageSnapshotRequest.startBarrageSnapshotRequest(metadata);
+        BarrageSnapshotRequest.addColumns(metadata, colOffset);
+        BarrageSnapshotRequest.addViewport(metadata, vpOffset);
+        BarrageSnapshotRequest.addSnapshotOptions(metadata, optOffset);
+        BarrageSnapshotRequest.addTicket(metadata, ticOffset);
+        BarrageSnapshotRequest.addReverseViewport(metadata, reverseViewport);
```

Comment:
What kinds of backwards compatibility do we have here?  Are old clients (0.35.2 being the first we might care about) going to be able to connect to new servers and fetch a snapshot or subscribe?

Reply:
I sure hope so. Flatbuffers are forward/backward compatible. Old clients should be able to request anything they already knew how to decode since this mapping change is focused on new mappings.